integration/binary_diff.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+# Basic binary compatibility test script for um2nc-standalone
+#
+# This script runs some basic tests to ensure binary compatibility
+# between Martix Dix's base script & changes introduced by the
+# um2nc-standalone development effort.
+#
+# TODO: add larger inputs with more vars to the testing?
+#
+#
+# Terminal session setup
+# $ export UM2NC_PROJ=<um2nc-standalone git dir>
+# $ export UM2NC_TEST_DATA=<dir with test netCDFs>
+#
+# Usage:
+# cd <um2nc-standalone git dir>
+# ./integration/binary_diff.sh
+#
+# NB: will display some um2nc output
+#
+# Script warns if there is a binary difference comparing netCDF files
+# If debugging a diff, the original source netCDF & the um2nc dev netCDF
+# can be converted to text with `ncks` & compared with a diff tool.
+
+echo "Binary equivalence/backwards compatibility diff for um2nc-standalone"
+
+if [ -z ${UM2NC_PROJ+x} ]; then
+  echo "ERROR: set UM2NC_PROJ to um2nc-standalone project dir";
+  exit
+fi
+
+if [ -z ${UM2NC_TEST_DATA+x} ]; then
+  echo "ERROR: set UM2NC_TEST_DATA to um2nc-standalone test data dir";
+  exit
+fi
+
+if [ ! -d "$UM2NC_TEST_DATA" ]; then
+  echo "ERROR: UM2NC_TEST_DATA dir does not exist";
+  exit
+fi
+
+function diff_warn {
+  # compare & warn if binary args don't match
+  diff -q "$1" "$2" | grep -E "^Files [a-zA-Z0-9. ]+differ"
```

Comment:
```suggestion
  diff -q "$1" "$2" | grep -E "^Files .+differ"
```
on gadi the current grep command misses the output when the files differ, resulting in no output when it was supposed to warn:

```
Files /g/data/tm70/sw6175/development/um2nc-standalone/integration/test_data/aiihca.paa1jan.subset.nomask.orig.nc and /g/data/tm70/sw6175/development/um2nc-standalone/integration/test_data/nomask.nc differ
```

The suggested change works for me, but let me know if you think it looks alright.

Reply:
That's a simpler more flexible regex. My orig regex might fail on "special" characters in the filename???

I've updated in 4769dc3a.